src/components/Groups/Groups.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-unused-vars */
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGroups } from '../../Redux/Sports/groups';
+import { fetchSports } from '../../Redux/Sports/sportsList';
+import Group from './Group';
+import '../../style/groups.css';
+
+const Groups = () => {
+  const dispatch = useDispatch();
+  const groups = useSelector((state) => state.groups.entities);
+  const sports = useSelector((state) => state.sports.entities);
+  const filter = useSelector((state) => state.toFilter.toFilter);
+  const filteredGropus = groups.filter((group) => group.attributes.name[0] === filter);
+  const displayableGroups = filter ? filteredGropus : groups;
+
+  useEffect(() => {
+    if (groups.length === 0) {
+      dispatch(fetchGroups());
+    }
+    if (sports.length === 0) {
+      dispatch(fetchSports());
+    }
+  });
+
+  const renderGroups = (groupCollection) => groupCollection.map((groupInfo) => (
+    <Group
+      key={groupInfo.id}
+      name={groupInfo.attributes.name}
+      number={groupInfo.relationships.sports.data.length}
+      slug={groupInfo.attributes.slug}
+    />
+  ));
+
+  return (
+    <div className="d-flex flex-wrap group-container">{renderGroups(displayableGroups)}</div>
```

Comment:
[OPTIONAL]

- Your app is working nice just a little recommendation that if no data is present for a certain alphabet you can show that there is no data available to make sure the user will not wait for the data.

For now, I have pressed **K** and it has no data, it's showing like this.

![Screenshot from 2021-09-11 23-39-03](https://user-images.githubusercontent.com/22434660/132957542-0b973e00-c275-4e8b-9d22-0005b8bba1af.png)